packages/ui/src/components/Visualization/Canvas/FormV2/fields/BeanField.tsx

```diff
@@ -0,0 +1,148 @@
+  const beansEntity = useMemo(() => {
+    return beansHandler.getBeansEntity();
+  }, [beansHandler]);
+
+  const items: TypeaheadItem<string | undefined>[] =
+    beansEntity?.parent.beans.map((item) => ({
+      name: item.name ? (beansHandler.getReferenceFromName(item.name) ?? '') : '',
+      description: String(item.type),
+      value: String(item.name),
+    })) ?? [];
+
+  const selectedItem = useMemo(() => {
+    if (!value) {
+      return undefined;
+    }
+    // Object values are stringified. Double check later different approach.
+    if (typeof value === 'object') {
+      return {
+        value: JSON.stringify(value),
+        name: JSON.stringify(value),
+        description: '',
+      };
+    }
+    return items.find((item) => item.name === value) ?? { value: value, name: value, description: '' };
+  }, [value]);
+
+  const onItemChange = useCallback(
+    (item?: TypeaheadItem<string>) => {
+      onChange(item!.name);
+    },
+    [onChange],
+  );
+
+  const onCleanInput = useCallback(() => {
+    onChange('');
+  }, [onChange]);
+
+  const createNewWithNameValue = 'create-new-bean-with-name';
+  const id = `${propName}-popover`;
+
+  const onSelect = useCallback(
+    (value: string | undefined, filterValue: string | undefined) => {
+      if (value) {
+        if (value === createNewWithNameValue) {
+          setInputValue(filterValue ?? '');
+          setIsNewBeanModalOpen(true);
+        } else {
+          setInputValue('');
+          setIsNewBeanModalOpen(true);
+        }
+      }
+    },
+    [BeanField],
+  );
+
+  const handleCreateBean = useCallback(
+    (model: BeanFactory) => {
+      beansHandler.addNewBean(getSerializedModel(model as unknown as Record<string, unknown>));
+
+      const beanRef = beansHandler.getReferenceFromName(model.name);
+
+      setIsNewBeanModalOpen(false);
+      onChange(beanRef ?? '');
+      setInputValue(beanRef as string);
+    },
+    [beansEntity, camelResource, onChange, selectedItem, beansHandler],
+  );
+
+  const handleCancelCreateBean = useCallback(() => {
+    setInputValue(beanReference);
+    setIsNewBeanModalOpen(false);
+  }, [beanReference]);
+  const javaType = schema.format ? extractGroup('bean', schema.format) : '';
+
+  if (!schema) {
+    return <div>BeanField - Schema not defined</div>;
+  }
+  return (
+    <FormGroup
+      fieldId={propName}
+      label={`${schema.title}`}
+      isRequired={required}
+      labelHelp={
+        <Popover
+          id={id}
+          headerContent={
+            <p>
+              {schema.title} {`<${schema.type}>`}
+            </p>
+          }
+          bodyContent={<p>{schema.description}</p>}
+          footerContent={<p>Default: {schema.default?.toString() ?? 'no default value'}</p>}
+          triggerAction="hover"
+          withFocusTrap={false}
+        >
+          <FormGroupLabelHelp aria-label={`More info for ${schema.title} field`} />
+        </Popover>
+      }
+    >
```

Comment:
To simplify things, we can use the `FieldWrapper` component now: https://github.com/KaotoIO/kaoto/pull/1939/files#diff-97a7a63d7b98efd7b359e05c952f68a6df9967192fee1b85d55d6adb859a9cebR36-R43

```
    <FieldWrapper
      propName={propName}
      required={required}
      title={schema.title}
      type="string"
      description={schema.description}
      defaultValue={schema.default?.toString()}
    >
```